File: Algorithms/Search Sort/Algorithms/quickSort/quicksort.js
function index() {
  let arr = [6, 3, 67, 2, 5, 234];
  let len = arr.length;
  quick(arr, 0, len);
  arr.forEach(el => console.log(el));
}

function quick(arr, baixo, alto) {
  if (baixo < alto) {
    let pi = quickPartition(arr, baixo, alto);
    quick(arr, baixo, pi - 1);
    quick(arr, pi + 1, alto);
  }
}

function quickPartition(arr, baixo, alto) {
  let pivot = arr[alto];
  let i = baixo - 1;
  for (let j = baixo; j < alto; j++){
    if (arr[j] < pivot) {
      i++;
      let temp = arr[i];
      arr[i] = arr[j]
      arr[j] = temp;
    }
  }
  i++;
  let temp = arr[i];
  arr[i] = arr[alto];
  arr[alto] = temp;
  i--;
  return (i + 1);
}
index();

File: Algorithms/Search Sort/Algorithms/quickSort/Quicksort.java

public class Quicksort {

  public static int quickPartition(int[] arr, int low, int high) {
    int pivot = arr[high];
    int i = low - 1;
    for (int j = low; j < high; j++) {
      if (arr[j] <= pivot) {
        i++;
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
      }
    }
    int temp = arr[i + 1];
    arr[i + 1] = arr[high];
    arr[high] = temp;
    return (i + 1);
  }

  public static void quickSort(int[] arr, int low, int high) {
    if (low < high) {
      int pi = quickPartition(arr, low, high);
      quickSort(arr, low, pi - 1);
      quickSort(arr, pi + 1, high);
    }
  }

  public static void main(String[] args) {
    int[] arr = { 6, 48, 46, 87, 91, 36, 19, 57, 49, 14 };
    int len = arr.length;
    quickSort(arr, 0, len - 1);
    for (int i : arr) {
      System.out.println(i + " ");
    }
  }
}
